From: Christopher Huhn <C.Huhn@gsi.de>
Date: Wed, 31 Oct 2018 16:04:57 +0100
Subject: Also select NS records from child domains

---
 build_bind.inc.php | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/build_bind.inc.php b/build_bind.inc.php
index e82ce0c..98bbdb8 100644
--- a/build_bind.inc.php
+++ b/build_bind.inc.php
@@ -347,9 +347,14 @@ EOF
 
     // loop through records and display them
     $q="
-    SELECT  *
-    FROM    dns
-    WHERE   domain_id = {$domain['id']}
+    SELECT  dns.*
+      FROM    dns, domains dom
+      WHERE   dns.domain_id = dom.id
+        AND ( domain_id = {$domain['id']}
+          OR ( dom.parent_id = {$domain['id']}
+            AND dns.type = 'NS'
+          )
+        )
     ORDER BY type";
 
 
