From: Christopher Huhn <C.Huhn@gsi.de>
Date: Wed, 31 Oct 2018 16:17:47 +0100
Subject: Beautify sort order

---
 build_bind.inc.php | 18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/build_bind.inc.php b/build_bind.inc.php
index 1335e43..f216690 100644
--- a/build_bind.inc.php
+++ b/build_bind.inc.php
@@ -347,8 +347,10 @@ EOF
 
     // loop through records and display them
     $q="
-    SELECT  dns.*
-      FROM    dns, domains dom
+    SELECT      dns.*
+      FROM      domains dom, dns
+      LEFT JOIN interfaces i
+        ON    dns.interface_id = i.id
       WHERE   dns.domain_id = dom.id
         AND ( domain_id = {$domain['id']} -- direct zone entries
           OR ( -- NS records of subdomains:
@@ -364,9 +366,15 @@ EOF
             ) AND domain_id <> {$domain['id']}
           )
         )
-    ORDER BY type";
-
-
+      ORDER BY CASE
+            WHEN type = 'NS'    THEN 1
+            WHEN type = 'MX'    THEN 2
+            WHEN type = 'SRV'   THEN 3
+            WHEN type = 'TXT'   THEN 4
+            WHEN type = 'A'     THEN 10
+            WHEN type = 'CNAME' THEN 11
+            WHEN type = 'PTR'   THEN 90
+        END, dns.name, ip_addr";
 
     // exectue the query
     $rs = $onadb->Execute($q);
